WireAuthentication/Sources/WireAuthenticationUI/Views/Login/LoginOrRegisterViaEmail/LoginOrRegisterViaEmailViewModel.swift

```diff
@@ -16,10 +16,53 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
-import Combine
 import Foundation
-import SwiftUI
 import WireAuthenticationAPI
+import WireReusableUIComponents
 
 @MainActor
-public final class LoginOrRegisterViaEmailViewModel: ObservableObject {}
+public final class LoginOrRegisterViaEmailViewModel: ObservableObject {
+
+    let router: any Router
+    let loginViaEmailUseCase: any LoginViaEmailUseCaseProtocol
+    let email: String
+    let forgotPasswordURL: URL
+    let passwordValidator: any PasswordValidator
+
+    // MARK: - Life cycle
+
+    public init(
+        router: any Router,
+        loginViaEmailUseCase: any LoginViaEmailUseCaseProtocol,
+        email: String,
+        accountsURL: URL,
+        passwordValidator: any PasswordValidator
+    ) {
+        self.router = router
+        self.loginViaEmailUseCase = loginViaEmailUseCase
+        self.email = email
+        self.forgotPasswordURL = accountsURL.appendingPathComponent("forgot")
+        self.passwordValidator = passwordValidator
+    }
+
+    func isValidPassword(_ password: String) -> Bool {
+        passwordValidator.validate(password)
+    }
+
+    func submitPassword(_ password: String) {
+        Task {
```

Comment:
suggestion: run this in a detached task, otherwise it will run on the main thread and block the UI.

Reply:
done